### ISSUE TITLE
Stack Level too deep (SystemStackError)

### ISSUE THREAD

**Issue by polytronix** [rails/rails#33062]
### Steps to reproduce
`rails plugin install git://github.com/ryanb/cancan.git`
`rails g cancan:ability`

### Expected behavior
Tell us what should happen

This should create a ability file for cancan in my ror app

### Actual behavior

`/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:90:` `warning:` `already` initialized constant FileUtils::VERSION
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:92: warning: previous definition of VERSION was here
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:1188: warning: already initialized constant FileUtils::Entry_::S_IF_DOOR
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1267: warning: previous definition of S_IF_DOOR was here
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:1446: warning: already initialized constant FileUtils::Entry_::DIRECTORY_TERM
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1541: warning: previous definition of DIRECTORY_TERM was here
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:1448: warning: already initialized constant FileUtils::Entry_::SYSCASE
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1543: warning: previous definition of SYSCASE was here
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:1501: warning: already initialized constant FileUtils::OPT_TABLE
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1596: warning: previous definition of OPT_TABLE was here
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:1555: warning: already initialized constant FileUtils::LOW_METHODS
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1650: warning: previous definition of LOW_METHODS was here
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:1562: warning: already initialized constant FileUtils::METHODS
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1657: warning: previous definition of METHODS was here
WARNING: Nokogiri was built against LibXML version 2.9.4, but has dynamically loaded 2.8.0
/usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:121: warning: constant ::Fixnum is deprecated
/usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:121: warning: constant ::Bignum is deprecated
Traceback (most recent call last):
	5368: from bin/rails:4:in `<main>'
	5367: from bin/rails:4:in `require'
	5366: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-4.2.1/lib/rails/commands.rb:17:in `<top (required)>'
	5365: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	5364: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:50:in `generate'
	5363: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:128:in `generate_or_destroy'
	5362: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:142:in `require_application_and_environment!'
	5361: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-4.2.1/lib/rails/application.rb:328:in `require_environment!'
	 ... 5356 levels...
	   4: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
	   3: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
	   2: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
	   1: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
/usr/local/lib/ruby/gems/2.5.0/gems/activesupport-4.2.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>': stack level too deep (SystemStackError)

### System configuration
**Rails version**:
Rails 4.2.1
**Ruby version**:
Ruby 2.5.1p57

**Comment by ahorek** [comment]
Your Rails version is quite old, Ruby 2.5 support Rails 4.2.8+, the current stable is Rails 5.2.
upgrade it with **gem upgrade rails**

**Comment by polytronix** [comment]
gem update rails

Rails 4.2.8

**Comment by louim** [comment]
You are also using a gem that hasn't been updated since 2013. You should probably be using the maintained version: https://github.com/CanCanCommunity/cancancan

**Comment by nynhex** [comment]
👍 This is not a Rails issue, any compatibility issues should be opened on the proper repo.

**Comment by polytronix** [comment]
I installed CANCANCAN but now I'm getting 

undefined method `timeout=' for Rack::Timeout:Class (NoMethodError)

**Comment by foxmuldercp** [comment]
seems that problem may be in fileutils?
in ruby 2.5.1 with clean rails 5.2 app have a same errors
[my bundler issue comment](https://github.com/bundler/bundler/issues/6540#issuecomment-396582520)

**Comment by charlietag** [comment]
@foxmuldercp I have the same issue.  I think it's ruby 2.5 issue. fileutils move to gem. originally it's under ruby stdlib.
<pre>
$ gem list |grep fileutils
fileutils (1.1.0, default: 1.0.2)
</pre>

``$ gem uninstall fileutils``
will fix this issue.
But I would like my gem update to latest version. 

Anyone has any idea??

**Comment by sikachu** [comment]
From the original stack trace, it looks like the problem is on this line: 

https://github.com/rails/rails/blob/v4.2.1/activesupport/lib/active_support/core_ext/numeric/conversions.rb#L131

That was probably because Ruby 2.4 unifies Fixnum and Bignum into Integer. I saw @jeremy's patch it here https://github.com/rails/rails/pull/25056/files#diff-38dfda2d84e271a6181fee7d0cd4ebd1L137 but it seems like that code had since moved on from 4.2.1's version already.

So, you could try to monkeypatch that code to make it support Ruby 2.5, or just use older version of Ruby with it. However, I'd rather suggest you to see if you can upgrade to Rails 5.2 and Ruby 2.5 instead as those are the current supported versions which should contain fixes for security vulnerabilities.

**Comment by charlietag** [comment]
@sikachu 
Ruby 2.5.1 + Rails 5.2.0 ===> will cause the same warning
<pre>
/usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/fileutils.rb:90:` `warning:` `already` initialized constant FileUtils::VERSION
</pre>

But yes, there are just warning. The application still works....

Just don't like the warning displayed, every time I do 'rails g something'....
<pre>
# ruby -v
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
</pre>

<pre>
/usr/local/lib/ruby/2.5.0/fileutils.rb:90: warning: already initialized constant FileUtils::VERSION
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:92: warning: previous definition of VERSION was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1188: warning: already initialized constant FileUtils::Entry_::S_IF_DOOR
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1267: warning: previous definition of S_IF_DOOR was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1446: warning: already initialized constant FileUtils::Entry_::DIRECTORY_TERM
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1541: warning: previous definition of DIRECTORY_TERM was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1448: warning: already initialized constant FileUtils::Entry_::SYSCASE
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1543: warning: previous definition of SYSCASE was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1501: warning: already initialized constant FileUtils::OPT_TABLE
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1596: warning: previous definition of OPT_TABLE was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1555: warning: already initialized constant FileUtils::LOW_METHODS
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1650: warning: previous definition of LOW_METHODS was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1562: warning: already initialized constant FileUtils::METHODS
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1657: warning: previous definition of METHODS was here
Rails 5.2.0
</pre>

<pre>
# rails -T
/usr/local/lib/ruby/2.5.0/fileutils.rb:90: warning: already initialized constant FileUtils::VERSION
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:92: warning: previous definition of VERSION was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1188: warning: already initialized constant FileUtils::Entry_::S_IF_DOOR
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1267: warning: previous definition of S_IF_DOOR was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1446: warning: already initialized constant FileUtils::Entry_::DIRECTORY_TERM
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1541: warning: previous definition of DIRECTORY_TERM was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1448: warning: already initialized constant FileUtils::Entry_::SYSCASE
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1543: warning: previous definition of SYSCASE was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1501: warning: already initialized constant FileUtils::OPT_TABLE
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1596: warning: previous definition of OPT_TABLE was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1555: warning: already initialized constant FileUtils::LOW_METHODS
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1650: warning: previous definition of LOW_METHODS was here
/usr/local/lib/ruby/2.5.0/fileutils.rb:1562: warning: already initialized constant FileUtils::METHODS
/usr/local/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:1657: warning: previous definition of METHODS was here
rails about                              # List versions of all Rails frameworks and the environment
rails active_storage:install             # Copy over the migration needed to the application
...
</pre>

**Comment by charlietag** [comment]
I'm using RVM to install ruby now. and no issues like above appears anymore...

**Comment by charveasna** [comment]
@charlietag, it is actually still the issue, i've just tried.

**Comment by charlietag** [comment]
@charveasna  strange. Mine ruby2.5.1(rvm) is working fine. no warning anymore....
<pre>
# gem list |grep fileutils
fileutils (1.1.0, default: 1.0.2)

# ruby -v
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]

# which ruby
~/.rvm/rubies/ruby-2.5.1/bin/ruby

# rails -v
Rails 5.2.0
</pre>

<pre>
# rails -T
rails about                              # List versions of all Rails frameworks and the environment
rails active_storage:install             # Copy over the migration needed to the application
rails active_storage:install:migrations  # Copy migrations from active_storage to application
rails app:template                       # Applies the template supplied by LOCATION=(/path/to/template) or URL
rails app:update                         # Update configs and some other initially generated files (or use just update:configs or update:bin)
rails assets:clean[keep]                 # Remove old compiled assets
rails assets:clobber                     # Remove compiled assets
rails assets:environment                 # Load asset compile environment
rails assets:precompile                  # Compile all the assets named in config.assets.precompile
rails cache_digests:dependencies         # Lookup first-level dependencies for TEMPLATE (like messages/show or com
.....
</pre>

**Comment by charlietag** [comment]
@charveasna  You're right... issue still exists.... no matter how I install ruby 2.5.1....

**Comment by kev9504** [comment]
I have the same issue with one of my recently created workspaces. Since I created it for tests, I performed the setup trying different stuff. The errors showed up as soon as I created my first application but did not affect the functionality at all, so I decided to created a new workspace with the usual setup and now the error is gone. 
  
`rvm install ruby-2.5.1 ` 
`rvm use-default ruby-2.5.1`
`gem install bundle`
`gem install rails`

I guess just creating a new workspace is not the best approach but it might help to isolate the issue, and it might be related to the initial setup.
